Mathlib/Tactic/SwapVar.lean

```diff
@@ -0,0 +1,28 @@
+/-
+Copyright (c) 2022 Arthur Paulino. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino
+-/
+
+import Mathlib.Util.Tactic
+
+namespace Mathlib.Tactic
+
+/--
+`swap_var swap_rule₁, swap_rule₂, ⋯` applies `swap_rule₁` then `swap_rule₂` then `⋯`.
+
+A *swap_rule* is of the form `x y` or `x ↔ y`, and "applying it" means swapping the variable name
+`x` by `y` and vice-versa on all hypotheses and the goal.
+
+```lean
+example {P Q : Prop} (q : P) (p : Q) : P ∧ Q := by
+  swap_var p ↔ q
+  exact ⟨p, q⟩
+```
+-/
+elab "swap_var " vars:(colGt (ident " ↔ "? ident)),+ : tactic =>
```

Comment:
```suggestion
syntax swapRule := ident " ↔ "? ident
elab "swap_var " vars:(colGt swapRule),+ : tactic =>
```
And then you can pattern match on it with ``let `(swapRule| $n1 $[↔]? $n2) := stx | unreachable!`` in the body.  Please avoid direct syntax accesses with `stx.raw[0][2]`, etc.  These are really awful to maintain and come with no static kind checks.

Reply:
Lean does not accept `let `(swapRule| $n1 $[↔]? $n2) := stx | unreachable!`